cardano-testnet/src/Testnet/Property/Util.hs

```diff
@@ -60,18 +63,44 @@ integrationWorkspace workspaceName f = withFrozenCallStack $
 isLinux :: Bool
 isLinux = os == "linux"
 
-
 -- | Runs an action in background, and registers cleanup to `MonadResource m`
--- The argument forces IO monad to prevent leaking of `MonadResource` to the child thread
+-- Concurrency is tricky in the 'ResourceT' monad. See the "Concurrency" section of
+-- https://www.fpcomplete.com/blog/understanding-resourcet/.
 runInBackground :: MonadTest m
                 => MonadResource m
-                => MonadCatch m
-                => IO a
+                => IO ()
+                -> IO a
                 -> m ()
-runInBackground act = void . H.evalM $ allocate (H.async act) cleanUp
-  where
-    cleanUp :: H.Async a -> IO ()
-    cleanUp a = H.cancel a >> void (H.link a)
+runInBackground runOnException act  =
```

Comment:
1. This is true, removed.
2 + 3: We can and should use `bracket` :+1: . Initially I was trying to share the file handle and so I reached for `resourceForkWith`. However we are simply waiting for another thread to be finished with the resource (i.e we are not interleaving).
4. In our use case when the parent thread completes our background process throws an exception because the node it was connected too shuts down. In the context of how we use `runInBackground` we can fork and forget because an exception will be thrown from `foldEpochState`. 
5. This is correct. I didn't have any specific scenarios in mind but if an unexpected exception is thrown we shouldn't ignore it.
6. Hmm I'm not sure. I think we want to be made aware of unexpected asynchronous exceptions here. There may be a scenario we will end up ignoring that we don't want to ignore.

Reply:
I responded in slack. This currently works, lets get it merged and you can investigate using `bracket` in a follow up PR.